### ISSUE TITLE
Build break: no-async

### ISSUE THREAD

**Issue by matbech** [openssl/openssl#707]
The build breaks with no-async
ssl/ssl_locl.h(1177): error C2061: syntax error: identifier 'ASYNC_JOB'

So maybe the ifndef is missing here:
```
#  ifndef OPENSSL_NO_ASYNC
    /* Async Job info */
    ASYNC_JOB *job;
#endif
```
But more is required. async.c is included in the build.

**Comment by mattcaswell** [comment]
Hmmm. That's wrong. The cause of the issue is this change as part of 1288f26fb9:

````
diff --git a/include/openssl/async.h b/include/openssl/async.h
index 8ec9b00..99d8075 100644
--- a/include/openssl/async.h
+++ b/include/openssl/async.h
@@ -53,6 +53,12 @@
 #ifndef HEADER_ASYNC_H
 # define HEADER_ASYNC_H
 
+#include <openssl/opensslconf.h>
+
+#ifdef OPENSSL_NO_ASYNC
+#define ASYNC_block_pause() do { ; } while(0)
+#define ASYNC_unblock_pause() do { ; } while(0)
+#else
 #include <stdlib.h>
 
 #if defined(_WIN32)
@@ -116,4 +122,5 @@ void ERR_load_ASYNC_strings(void);
 #ifdef  __cplusplus
 }
 #endif
+#endif /* OPENSSL_NO_ASYNC */
 #endif
````
That change needs reverting. I'm not sure why that change was deemed necessary because no-async seems to work fine (for me) without that, but fails with it! Yet, the commit description where that change was introduced suggests it was done to fix no-async??

**Comment by mattcaswell** [comment]
Fixed in commit 316fae2a
